localstack/services/s3/utils.py

```diff
@@ -151,6 +156,92 @@ def verify_checksum(checksum_algorithm: str, data: bytes, request: Dict):
         )
 
 
+def decode_aws_chunked_object(
+    stream: IO[bytes],
+    buffer: IO[bytes],
+    content_length: int,
+    checksum_algorithm: str = None,
+    checksum_value: str = None,
+) -> tuple[str, str]:
+    """
+
+    :param stream: the original stream to read
+    :param buffer: the stream where we set the decoded data
+    :param content_length: the total length of the original stream
+    :param checksum_algorithm:
+    :param checksum_value:
+    :return:
+    """
```

Comment:
Would be great to have a better docstring here. How does the return tuple look like? What's the algo and value params?